### DISCUSSION TITLE
Tests and next/router

### DISCUSSION THREAD

**Discussion by A11yship** [neuefische/questions#110] [Web] [answered]
Hello,

I create a new page for the form and use  `router.push('/');` in the submit event to go back to the index page. That works fine, but the tests fail wit the error: `[TypeError: Cannot read properties of null (reading 'push')]`. I guess I need to mock `next/router`. How do I do that? I don't find an example for that.

My current [PR](https://github.com/A11yship/capstone-project/pull/19) and the [deployment](https://capstone-project-git-formpage-a11yship.vercel.app/).

Thanks for your help.

**Comment by alexclaes** [comment, marked as answer]
Hello Hannah,

yes! Adding a mock for `next/router` is the right approach. There are several ways on how to do this. 

A very short and simple solution is to create mock within the `.spec.js` file for exactly the information/methods that your component uses from the router.

Try adding this code after all imports, but before the `describe` block.

```js
jest.mock('next/router', () => ({
	useRouter() {
		return {
			push: jest.fn(),
		};
	},
}));
```

**Reply by A11yship** [reply]
That works. Thank you :).
